### ISSUE TITLE
Dear author, could you share the code and script on the M4 dataset. esteem it  a favor

### ISSUE THREAD

**Issue by 2ySong** [VEWOXIC/FITS#20]

**Comment by 2ySong** [comment]
And your F1 score in the anomaly detection task. After I run it, it displays 
`Accuracy : 0.8234, Precision : 0.9677, Recall : 0.2300, F-score : 0.3716 `. Instead of the 78.12 reported in the paper, I am a beginner and therefore have doubts. Thank you for your reply.

**Comment by VEWOXIC** [comment]
I will check the M4 script, and if my memory is correct I directly put the FITS model to a repo that has already implement the M4,M5 dataloader for running.

**Comment by VEWOXIC** [comment]
As for the AD task, there should be a separate threshold tuning process after the training. maybe you can check.

**Comment by 2ySong** [comment]
Thank you very much for your reply. I have one more question about calculating the number of parameters in FITS. I saw someone on OpenReview mention that complex numbers are counted as double the parameter count, but when I use Lightning or other scripts to calculate the parameter count, it comes out as single.

**Comment by 2ySong** [comment]
> 至于AD任务，训练后应该有一个单独的阈值调整过程。也许你可以检查一下。

`dataset: MSL
individual: False
input_c: 55
k: 3
lr: 0.0001
mode: test
model_save_path: checkpoints
num_epochs: 10
output_c: 55
plot: False
pretrained_model: 20
win_size: 400
-------------- End ----------------
test: (73729, 1)
train: (58317, 1)
test: (73729, 1)
train: (58317, 1)
test: (73729, 1)
train: (58317, 1)
test: (73729, 1)
train: (58317, 1)
Model(
  (freq_upsampler): Linear(in_features=50, out_features=200, bias=True)
)
======================TEST MODE======================
Threshold : 0.10661790966987611
(73330,) (73330,)
pred:    (73330,)
gt:      (73330,)
pred:  (73330,)
gt:    (73330,)
Accuracy : 0.7846, Precision : 0.6892, Recall : 0.6438, F-score : 0.6657 
` This is my run results. I can't understand the correlation between `Accuracy : 0.7846, Precision : 0.6892, Recall : 0.6438, F-score : 0.6657 ` and 
![image](https://github.com/user-attachments/assets/16d881a1-5264-4251-84d4-4731ddf7e61a)
. Thanks.

**Comment by VEWOXIC** [comment]
I am not sure on which dataset you are experimenting on. And there should be an anomaly rate tuning process as [this](https://github.com/VEWOXIC/FITS/blob/001738e60af6d52f3efdc964a2a82942d1304c05/AD/scripts/SMAP.sh#L22) to get the correct threshold.

**Comment by 2ySong** [comment]
> 我不确定您正在对哪个数据集进行实验。应该有一个异常率调整过程来获得正确[的](https://github.com/VEWOXIC/FITS/blob/001738e60af6d52f3efdc964a2a82942d1304c05/AD/scripts/SMAP.sh#L22)阈值。

yeal，I follow your script in MSL dataset. I got 20 logs after test.
![image](https://github.com/user-attachments/assets/ae55d1c1-787c-41eb-8d3e-9e9dfe40dac1)
